standardtracer/propagation.go

```diff
@@ -44,9 +45,11 @@ func (p *splitTextPropagator) InjectSpan(
 		fieldNameSampled: strconv.FormatBool(sc.raw.Sampled),
 	}
 	sc.Lock()
-	splitTextCarrier.TraceAttributes = make(map[string]string, len(sc.raw.Attributes))
-	for k, v := range sc.raw.Attributes {
-		splitTextCarrier.TraceAttributes[k] = v
+	if l := len(sc.raw.Attributes); l > 0 {
+		splitTextCarrier.TraceAttributes = make(map[string]string, l)
```

Comment:
Header is a map, but you're right it's the wrong map, with multiple values per key:
```
type Header map[string][]string
```
But Go native http is not the only protocol, other protocols may have headers as plain map, so I think the principle holds.

I actually wanted one of the standard injector carriers to be an interface like
```
type SplitTextCarrier interface {
    SetTracerContextField(key, value string)
    SetTraceAttributeField(key, value string)
}
```